lib/improver/standardise.py

```diff
@@ -0,0 +1,400 @@
+
+        Returns:
+            iris.cube.Cube
+        """
+        coords_to_collapse = []
+        for coord in cube.coords(dim_coords=True):
+            if len(coord.points) == 1 and "realization" not in coord.name():
+                coords_to_collapse.append(coord)
+        for coord in coords_to_collapse:
+            cube = next(cube.slices_over(coord))
+        return cube
+
+    @staticmethod
+    def _remove_scalar_coords(cube, coords_to_remove):
+        """Removes named coordinates from the cube."""
+        for coord in coords_to_remove:
+            try:
+                cube.remove_coord(coord)
+            except CoordinateNotFoundError:
+                continue
+
+    def process(self, cube, target_grid=None, new_name=None,
+                new_units=None, coords_to_remove=None, attributes_dict=None,
+                fix_float64=False):
+        """
+        Perform regridding and metadata adjustments
+
+        Args:
+            cube (iris.cube.Cube):
+                Input cube to be standardised
+            target_grid (iris.cube.Cube or None):
+                Cube on the required grid. For "nearest-with-mask" regridding,
+                this cube should contain a binary land-sea mask
+                ("land_binary_mask").
+            new_name (str or None):
+                Optional rename for output cube
+            new_units (str or None):
+                Optional unit conversion for output cube
+            coords_to_remove (list of str or None):
+                Optional list of scalar coordinates to remove from output cube
+            attributes_dict (dict or None):
+                Optional dictionary of required attribute updates. Keys are
+                attribute names, and values are the required value or "remove".
+            fix_float64 (bool):
+                Flag to de-escalate float64 precision
+
+        Returns:
+            iris.cube.Cube
+        """
+        # regridding
+        if target_grid:
+            # if regridding using a land-sea mask, check this is provided on
+            # the source grid
+            if self.REGRID_REQUIRES_LANDMASK[self.regrid_mode]:
+                if not spatial_coords_match(cube, self.landmask_source_grid):
+                    raise ValueError(
+                        "Source landmask does not match input grid")
+            cube = self._regrid_to_target(cube, target_grid)
+
+        # standard metadata updates
+        cube = self._collapse_scalar_dimensions(cube)
+        self._standardise_time_coordinates(cube)
+
+        # optional metadata updates
+        if new_name:
+            cube.rename(new_name)
+        if new_units:
+            cube.convert_units(new_units)
+        if coords_to_remove:
+            self._remove_scalar_coords(cube, coords_to_remove)
+        if attributes_dict:
+            amend_attributes(cube, attributes_dict)
+
+        check_cube_not_float64(cube, fix=fix_float64)
+
+        return cube
+
+
+class RegridLandSea:
```

Comment:
Plugin moved without changes from `improver.utilities.spatial`.  Since this is the only place it is, and should be, called, because of the very custom processing and the need to wrap the `process` call with external error-handling.  The associated unit test file has also been moved without modification (import redirection only).